File: src/main/java/org/shark/example/config/swagger/SwaggerConfig.java
package org.shark.example.config.swagger;

import io.swagger.v3.oas.models.info.Info;
import org.springdoc.core.models.GroupedOpenApi;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SwaggerConfig {

    @Bean
    GroupedOpenApi multiLanguageExampleApis() {
        return GroupedOpenApi.builder()
                .group("example-apis")
                .displayName("Example Apis")
                .packagesToScan("org.shark.example.controller")
                .addOpenApiCustomizer(openApi ->
                        openApi.info(info("Example APIS", "Example APIS")))
                .build();
    }

    private Info info(String title, String description) {
        return new Info()
                .title(title)
                .version("0.0.1")
                .description(description);
    }

}

File: src/main/kotlin/org/shark/example/config/swagger/SwaggerConfig.kt
package org.shark.example.config.swagger

import io.swagger.v3.oas.models.info.Info
import org.springdoc.core.models.GroupedOpenApi
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SwaggerConfig {

    @Bean
    fun multiLanguageExampleApis(): GroupedOpenApi {
        return GroupedOpenApi.builder()
            .group("emqx-example-apis")
            .displayName("Emqx Example Apis")
            .packagesToScan("org.shark.example.controller.emqx")
            .addOpenApiCustomizer { openApi ->
                openApi.info(info("EMQX Example APIS", "EMQX Example APIS"))
            }
            .build()
    }

    private fun info(title: String, description: String): Info {
        return Info()
            .title(title)
            .version("0.0.1")
            .description(description)
    }
}
